Source/FullScreenPokemon.ts

```diff
@@ -1222,13 +1222,14 @@ module FullScreenPokemon {
 
             for (let i: number = 0; i < characters.length; i += 1) {
                 character = characters[i];
-                FSP.shiftCharacter(character);
 
-                if (character.shouldWalk && !FSP.MenuGrapher.getActiveMenu()) {
-                    character.onWalkingStart(character, character.direction);
+                if ((character.shouldWalk || character.forceWalk) && !FSP.MenuGrapher.getActiveMenu()) {
```

Comment:
I'm not a fan of adding this much logic to `maintainCharacters` just for a rarely-used feature.

Proposal: Set `shouldWalk` on walking end based on `forceWalk` so you don't have to modify the if statement here.